stm_workshop/lib/utils/timer.cpp

```diff
@@ -0,0 +1,56 @@
+#include "timer.hpp"
+
+#include <sys/time.h>
+
+namespace hyped {
+namespace utils {
+
+uint64_t Timer::time_start_ = Timer::getTimeMicros();
+
+uint64_t Timer::getTimeMicros()
+{
+  timeval tv;
+  if (gettimeofday(&tv, (struct timezone *)0) < 0) { return 0; }
+  return (static_cast<uint64_t>(tv.tv_sec) * 1000000) + tv.tv_usec - time_start_;
+}
+
+Timer::Timer() : elapsed_(0), start_(0), stop_(0)
+{ /* EMPTY */
+}
+
+void Timer::start()
+{
+  start_ = getTimeMicros();
+  stop_  = 0;
+}
+
+void Timer::stop()
+{
+  stop_ = getTimeMicros();
+  elapsed_ += stop_ - start_;
```

Comment:
Idk if I'm missing something here but why `+=` here? Is it not just `=`?

Reply:
It's analogous to being able to pause a stopwatch and then pressing start again - ensuring the total elapsed time is always getting added on to.

Reply:
But then wouldn't you then have to reassign `start_` to be the point at which you paused the stopwatch otherwise you would be counting a single segment of time multiple times. Not sure if this has been done and I just missed it?

Reply:
The reassignment happens when `start()` is called, so the time between calling `stop()` and then `start()` again is not taken into account (it's not immediately reassigned for this reason).